### PR TITLE
chore(wallet): LedgerJS Breaking Changes

### PR DESCRIPTION
**Pull request by tomislavhoracek**
# Context
We need to update `ledgerjs-hw-app-cardano` to the latest published version `5.0.0.`. Update has few breaking changes and this PR will cover them.

# Proposed Solution
This task is for updating `cardano-js-sdk's` Ledger support implementation with respect to the breaking changes in the `ledgerjs-hw-app-cardano` package.

# Important Changes Introduced
`ledgerjs-hw-app-cardano` updated to version `5.0.0.`

### CODE REVIEW

**Review by rhyslbw — DISMISSED** [review]
LGTM. Great work @tomislavhoracek

File: packages/wallet/test/e2e/config.ts [hunk 1]
```diff
@@ -145,7 +145,14 @@ export const keyAgentByIdx = async (accountIndex: number) => {
   }
 };
 
-export const keyAgentReady = (() => keyAgentByIdx(0))();
+export const keyAgentReady = (() => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const keyAgent = keyAgentByIdx(0) as any;
+  if (process.env.KEY_AGENT === 'Ledger' && keyAgent.deviceConnection) {
+    deviceConnection = keyAgent.deviceConnection;
```
Comment (mkazlauskas) [inline packages/wallet/test/e2e/config.ts:152]: Isn't this redundant as `keyAgentByIdx` sets it?

Comment (tomislavhoracek) [reply packages/wallet/test/e2e/config.ts:152]: Note: Potential solution will be to memoize keyAgentByIdx(0) like we discussed.

Comment (tomislavhoracek) [reply packages/wallet/test/e2e/config.ts]: ✅

**Review by mkazlauskas — APPROVED** [review]
Nice work!

**Pull request merged**